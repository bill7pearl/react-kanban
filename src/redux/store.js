import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import rocketReducer from './rockets/rockets';
import missionReducer from './missions/missions';

const store = configureStore(
  {
    reducer: {
      rockets: rocketReducer,
      missions: missionReducer,
    },
  },
  applyMiddleware(thunk),
);

export default store;
