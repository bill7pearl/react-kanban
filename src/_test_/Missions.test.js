import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Missions from '../components/Missions';

describe('Missions', () => {
  it('renders Missions component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Missions />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
});
