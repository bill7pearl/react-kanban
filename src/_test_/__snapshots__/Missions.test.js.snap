// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders Missions component 1`] = `
<main>
  <nav>
    <h4>
      Mission
    </h4>
    <h4>
      Description
    </h4>
    <h4>
      Status
    </h4>
    <button
      type="submit"
    >
      Join Mission
    </button>
  </nav>
  <div />
</main>
`;
